google/cloud/storage/internal/grpc_buffer_read_object_data.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/grpc_buffer_read_object_data.h"
+#include <algorithm>
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+
+std::size_t GrpcBufferReadObjectData::FillBuffer(char* buffer, std::size_t n) {
+  std::size_t offset = 0;
+  for (auto v : spill_view_.Chunks()) {
+    if (offset >= n) break;
```

Comment:
Opt: I thought you were on the weakest-loop-guard team ... so `if (offset == n) break`?

Reply:
Done